src/codec.cairo

```diff
@@ -3,6 +3,9 @@
 use super::types::transaction::{Transaction, TxIn, TxOut, OutPoint};
 use raito::utils::hash::Hash;
 
+const WITNESS_FACTOR: usize = 1;
+const NON_WITNESS_FACTOR: usize = 4;
+
 pub trait Encode<T> {
```

Comment:
I would either add another trait `Weight` since encoding is not weight or is we consider both to be related enough, somehow combine encoding and weight calculation. I mean encode_to would need to encode to a buffer struct which would hold weight and ByteArray. There would not be any weight method. @m-kus WDYT?

Reply:
You are right, it seems to overcomplicate things, let's just add `weight()` method to the `transaction::TransactionTrait`?

Reply:
Given that Encode and Weight are rather closely related, I've tried adding Weight to TransactionTrait, and it seems much cleaner.